cmake/modules/hunter_check_toolchain_definition.cmake

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2016 NeroBurner
+# All rights reserved.
+
+include(CMakeParseArguments) # cmake_parse_arguments
+
+include(hunter_test_string_not_empty)
+
+function(hunter_check_toolchain_definition)
+  set(function_name "hunter_check_toolchain_definition")
+  set(function_synopsis "${function_name}(NAME definition [DEFINED out_is_defined] [VALUE out_value])")
+
+  # parse arguments
+  set(one_value_args NAME DEFINED VALUE)
+  cmake_parse_arguments(hunter_ct_arg "" "${one_value_args}" "" ${ARGV})
+
+  string(COMPARE NOTEQUAL "${hunter_ct_arg_NAME}"    "" arg_NAME)
+  string(COMPARE NOTEQUAL "${hunter_ct_arg_DEFINED}" "" arg_DEFINED)
+  string(COMPARE NOTEQUAL "${hunter_ct_arg_VALUE}"   "" arg_VALUE)
+  # No free arguments allowed
+  list(LENGTH hunter_ct_arg_UNPARSED_ARGUMENTS hunter_ct_len)
+  if(NOT ${hunter_ct_len} EQUAL 0)
+    hunter_internal_error(
+      "'${function_name}' incorrect usage,"
+      " expected no free argument."
+      " Synopsis: ${function_synopsis}"
+    )
+  endif()
+  # option NAME is mandatory
+  if(NOT arg_NAME)
+    hunter_internal_error(
+      "'${function_name}' incorrect usage,"
+      " option 'NAME' with one argument is mandatory."
+      " Synopsis: ${function_synopsis}"
+    )
+  endif()
+  string(FIND "${hunter_ct_arg_NAME}" " " arg_NAME_whitespace_position)
+  if(arg_NAME_WHITESPACE_POSITION EQUAL -1)
```

Comment:
`arg_NAME_whitespace_position` and `arg_NAME_WHITESPACE_POSITION` are different variables.